platforms/nuttx/src/px4/common/hrt_ioctl.c

```diff
@@ -336,17 +351,24 @@ hrt_ioctl(unsigned int cmd, unsigned long arg)
 
 	case HRT_UNREGISTER: {
 			px4_sem_t *callback_sem = *(px4_sem_t **)arg;
+			sq_entry_t *queued;
 			struct usr_hrt_call *e;
 			irqstate_t flags;
 
 			flags = spin_lock_irqsave_wo_note(&g_hrt_ioctl_lock);
 
-			while ((e = (void *)sq_remfirst(&callout_queue))) {
+			sq_for_every(&callout_queue, queued) {
+				e = (struct usr_hrt_call *)queued;
+
 				if (callback_sem == e->entry.callout_sem) {
+					sq_rem(&e->list_item, &callout_queue);
 					hrt_cancel(&e->entry);
+
 					/* Remove potential inflight entry as well */
 					sq_rem(&e->list_item, &callout_inflight);
-					kmm_free(e);
```

Comment:
Is there some reason this was removed? It's not immediately obvious.

Reply:
I was thinking that it is not safe to do that inside of the loop, where the next iteration pointer is taken from the e which was just deleted.

The deletion should be unnecessary anyways, since it should not grow the maximum used kernel heap; the next allocation is anyhow taken from the freelist

Reply:
I can, of course, do the deletion as well in safe manner if you like!

Reply:
pushed a suggestion on top

Reply:
It is fine to me to not free the entries at all, was just wondering why the line was removed